flink-runtime/src/main/java/org/apache/flink/runtime/deployment/ResultPartitionDeploymentDescriptor.java

```diff
@@ -48,6 +50,18 @@
 	/** Flag whether the result partition should send scheduleOrUpdateConsumer messages. */
 	private final boolean sendScheduleOrUpdateConsumersMessage;
 
+	/** Whether the result partition is managed externally.
+	 *
+	 * <p>An externally managed partition is only released if either
+	 * a) the {@link ShuffleEnvironment} is instructed to do so via {@link ShuffleEnvironment#releasePartitions(Collection)},
+	 * b) the production of the partition fails,
+	 * c) the {@link ShuffleEnvironment} shuts down.
+	 *
+	 * <p>An internally managed partition may be released by the {@link ShuffleEnvironment} at any time it deems
+	 * appropriate, typically after it has been fully consumed once.
+	 * */
+	private final boolean isManagedExternally;
```

Comment:
I think it would be better to name this field `releaseOn/AfterConsumption` because it better describes what this field influences.